hooks/shapediver/appbuilder/useAppBuilderStandardContainers.tsx

```diff
@@ -0,0 +1,182 @@
+	IAppBuilder,
+	IAppBuilderContainer,
+	IAppBuilderSettingsResolved,
+	IAppBuilderSettingsSession,
+	isStandardContainer,
+} from "@AppBuilderShared/types/shapediver/appbuilder";
+import React, {useEffect, useMemo} from "react";
+import {useSessionPropsExport} from "../parameters/useSessionPropsExport";
+import {useSessionPropsOutput} from "../parameters/useSessionPropsOutput";
+import {useSessionPropsParameter} from "../parameters/useSessionPropsParameter";
+import {useViewportAnchors} from "../viewer/useViewportAnchors";
+
+interface Props {
+	namespace: string;
+	appBuilderData?: IAppBuilder;
+	defaultSessionDto?: IAppBuilderSettingsSession;
+	settings?: IAppBuilderSettingsResolved;
+	hasAppBuilderOutput: boolean;
+}
+
+/**
+ * Create rendering hints for the container.
+ * @param container
+ * @returns
+ */
+const createContainerHints = (
+	container: IAppBuilderContainer,
+): IAppBuilderTemplatePageContainerHints | undefined => {
+	// if the bottom container contains tabs, prefer vertical layout
+	if (
+		container.name === AppBuilderContainerNameType.Bottom &&
+		container.tabs &&
+		container.tabs.length > 0
+	) {
+		return {
+			preferVertical: true,
+		};
+	}
+};
+
+/**
+ * Hook to manage standard containers for the app builder.
+ * This hook provides the necessary props and state management for the standard containers.
+ * It also handles the viewport anchors.
+ *
+ * Once the mergedContainers are available, it creates the necessary UI elements for each container.
+ *
+ * @param props
+ * @returns
+ */
+export function useAppBuilderStandardContainers(props: Props) {
+	const {
+		namespace,
+		appBuilderData,
+		defaultSessionDto,
+		settings,
+		hasAppBuilderOutput,
+	} = props;
+
+	// get props for fallback parameters
+	const parameterProps = useSessionPropsParameter(namespace);
+	const exportProps = useSessionPropsExport(namespace);
+	const outputProps = useSessionPropsOutput(
+		namespace,
+		(output) => !!output.chunks,
+	);
+
+	const {mergedContainers, resetDefaultContainers, setDefaultContainer} =
+		useShapeDiverStoreStandardContainers((state) => ({
+			mergedContainers: state.mergedContainers,
+			resetDefaultContainers: state.resetDefaultContainers,
+			setDefaultContainer: state.setDefaultContainer,
+		}));
+
+	// viewport anchors
+	const anchors = useViewportAnchors({
+		namespace,
+		containers: appBuilderData?.containers,
+	});
+
+	// should fallback containers be shown?
+	const showFallbackContainers =
+		settings?.settings?.disableFallbackUi !== true;
+
+	// Update store with default containers whenever appBuilderData changes
+	// This only stores the container definition within the store
+	useEffect(() => {
+		// Reset containers first
+		resetDefaultContainers();
+
+		if (appBuilderData?.containers) {
+			// Set containers from appBuilderData
+			appBuilderData.containers.forEach((container) => {
+				if (isStandardContainer(container)) {
+					setDefaultContainer(container.name, container);
+				}
+			});
```

Comment:
Optimization idea: Likely we can implement a `setDefaultContainers` state function which does everything in a single `set` call in the store, thereby avoiding some re-rendering.